core/templates/services/math-interactions.service.ts

```diff
@@ -184,6 +184,35 @@ export class MathInteractionsService {
   }
 
   insertMultiplicationSigns(expressionString: string): string {
```

Comment:
Why are the changes below in `insertMultiplicationSigns` function? Also, why are the changes below required?
Since, it was working without these before, right?

Reply:
The problem was that the older implementation didn't account for consecutive greek words without multiplication signs so things like `alphabeta` would be considered as `a*l*p*h*a*b*e*t*a` instead of `alpha*beta`. With these changes, such cases are also covered. Also, cases like `xalphayzgamma` are converted to `x*alpha*y*z*gamma`.